File: src/main/java/nl/hanze/week2/ATM/IdleState.java
package nl.hanze.week2.ATM;

/**
 * Created by Laurens on 9-10-2017.
 */
public class IdleState implements ATMState {
    @Override
    public void insertCard(ATM atm) {
        System.out.println("Succesfully inserted card");
        atm.setState(new HasCard());
    }

    @Override
    public void ejectCard(ATM atm) {
        System.out.println("No card present");
    }

    @Override
    public void insertPin(ATM atm) {
        System.out.println("No card present");
    }

    @Override
    public void requestAmount(ATM atm) {
        System.out.println("No card present");
    }
}

File: src/main/java/nl/hanze/week2/assignment45/IdleState.kt
package nl.hanze.week2.assignment45

/**
 * Created by Laurens on 9-10-2017.
 */
class IdleState : ATMState {
    override fun insertCard(atm: ATM) {
        println("Succesfully inserted card")
        atm.state= HasCard()
    }

    override fun ejectCard(atm: ATM) = println("No card present")

    override fun insertPin(atm: ATM) = println("No card present")

    override fun requestAmount(atm: ATM) = println("No card present")
}
